src/nanotron/trainer.py

```diff
@@ -206,9 +206,12 @@ def __init__(
             self.consumed_train_samples = 0
 
         # Setup tensorboard write and log writers on output rank
-        self.logger_ranks = self.parallel_context.world_rank_matrix[
-            0, self.unwrapped_model.output_pp_rank, 0, 0
-        ].flatten()
+        self.logger_ranks = self.parallel_context.get_global_rank(
+            ep_rank=0,
+            pp_rank=self.unwrapped_model.output_pp_rank,
+            dp_rank=0,
+            tp_rank=0
+        ).flatten()
```

Comment:
Do we need the flatten()? Can we remove it too?

Reply:
I am not sure about that, because the return type of `get_global_rank` is `np.int64`. We use `.flatten()` here is that it converts the return type from a scalar `np.int64` into a `numpy.ndarray`. This conversion is crucial because, in other parts of the code, `self.logger_ranks` is expected to be an array.

Without using `.flatten()`, we could encounter an `IndexError` and a `TypeError` related to trying to iterate over or index into what is essentially a scalar (`numpy.int64`) value. This would require modifications in other parts of the code where `self.logger_ranks` is treated as an array.

https://github.com/huggingface/nanotron/blob/07caff1940609f09096cb4b0fd555441799ea252/src/nanotron/trainer.py#L235-L241

https://github.com/huggingface/nanotron/blob/07caff1940609f09096cb4b0fd555441799ea252/src/nanotron/trainer.py#L431-L432

and other 2 similar occurrences, we can see that `self.logger_ranks` is indeed used as an array.

Reply:
You're right! Thanks for the details explanation :)